synse/http.go

```diff
@@ -112,28 +117,222 @@ func (c *httpClient) Config() (*Config, error) {
 	return out, nil
 }
 
+// Plugins returns the summary of all plugins currently registered with
+// Synse Server.
+func (c *httpClient) Plugins() (*[]scheme.PluginMeta, error) {
+	out := new([]scheme.PluginMeta)
+	err := c.getVersioned(pluginURI, out)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to request `/plugin` endpoint")
+	}
```

Comment:
since all of these fns are calling a helper function (getVersioned, postVersioned, ...) to dispatch and handle the request, I'm not sure the error needs to be wrapped here. the helper fn is given the `pluginURI`, so it should be able to format that into the error response in a more generalized way. Once thats done, there is no additional info being wrapped here, so we can just return the error as-is, e.g. something like
```go
if err := c.getVersioned(pluginURI, out); err != nil {
	return nil, err
}
```

This doesn't need to be done in this PR, but if deferring for later we should open an issue for it for tracking.